### PR TITLE
Py3: Fix error handling in download plugin

### PR DESCRIPTION
**Pull request by jeroenl**
While trying to store an error page, the plugin returned:
`TypeError: write() argument must be str, not bytes`

### CODE REVIEW

File: flexget/plugins/output/download.py [hunk 1]
```diff
@@ -155,9 +155,9 @@ def save_error_page(self, entry, task, page):
         received = os.path.join(task.manager.config_base, 'received', task.name)
         if not os.path.isdir(received):
             os.makedirs(received)
-        filename = os.path.join(received, '%s.error' % entry['title'].encode(sys.getfilesystemencoding(), 'replace'))
+        filename = os.path.join(received, '%s.error' % entry['title'].encode(sys.getfilesystemencoding(), 'replace').decode(sys.getfilesystemencoding()))
```
Comment (stevezau) [inline flexget/plugins/output/download.py]: Hmm, we are encoding then decoding?? that does not sound right?

what error did you get?

Comment (jeroenl) [inline flexget/plugins/output/download.py]: The error came below. For this, I noticed that the filename shows up as:
path/to/b'entry title'.error
as opposed to (in Py2)
path/to/entry title.error

Encode then decode in this case would at least drop characters that are not valid in the file system encoding, which I'm guessing at some point fixed a bug. I do wonder though if maybe we shouldn't change the way the file name is generated. Perhaps using pathscrub?

**Comment by stevezau** [comment]
hmm yeah i think we can just do entry['title'] now as we enforce it to be text here https://github.com/Flexget/Flexget/blob/646d6ed86a79599ff1c9aba9107cadbccfddff77/flexget/entry.py#L189-L194

do you want to update the PR?

**Comment by gazpachoking** [comment]
Maybe we should use path.py for these filename operations rather than raw text

**Comment by jeroenl** [comment]
How's this?

**Comment by stevezau** [comment]
hmm looks ok to me.. @gazpachoking ?

**Comment by gazpachoking** [comment]
Looks fine to me.

**Comment by cvium** [comment]
@jeroenl Wanna sync this so we can merge?

**Pull request merged**